### PR TITLE
Verify NoW Contacts and Update CORE contacts

### PR DESCRIPTION
**Pull request by mmumby**
# MDS-3186
- Updated `VerifyNoWContacts` according to designs and new flow
Screen Shot 2021-02-19 at 1 46 27 PM
- Ability to Edit Core party information if it is incorrect
Screen Shot 2021-02-19 at 1 47 14 PM
- Ability to Search for a contact if it is not displayed in the list
- Ability to create a new CORE contact, the form is populated with the application contact
Screen Shot 2021-02-19 at 2 56 50 PM
- Disabled the Submit button until all contacts are verified

### CODE REVIEW

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 1]
```diff
@@ -1,28 +1,38 @@
+/* eslint-disable */
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:1]: 😁 ?

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 2]
```diff
@@ -48,211 +87,264 @@ const renderContacts = ({
...
-                          ) : (
-                            <p>{Strings.EMPTY_FIELD}</p>
-                          )}
+      <Col span={8}>
+        <Row className="contact-rows">
+          <Col span={24} style={{ minHeight: "130px" }}>
+            <h3>Application Contacts</h3>
+            <p>
+              Contacts listed here come from the original Notice of Work. Click &quot;Search
+              Contact&quot; to see a list of possible CORE matches in the &quot;Matching contact
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:105]: "CORE" --> "Core"? Why all caps?
"Matching contact options" --> "Matching Contact Options"

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 3]
```diff
@@ -48,211 +87,264 @@ const renderContacts = ({
...
+            const selectedCoreParty = selectedData
+              .filter(({ value }) => selectedCorePartyGuid === value)
+              .map((contact) => contact)[0];
+            const contactInformation = selectedCoreParty ? selectedCoreParty : fields.get(index);
+            return (
+              <Col span={24} key={fields.get(index).id}>
+                <Card
+                  hoverable
+                  style={
+                    isSelectedContact ? { boxShadow: "rgb(124 102 173) 0px 0px 10px 3px" } : {}
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:129]: Should we add this to the `ant-card-now` class?
```
&-selected {
...
}
```

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 4]
```diff
@@ -48,211 +87,264 @@ const renderContacts = ({
...
+                          : "Contact"
+                      }`}</span>
+                      {!confirmedContacts?.includes(fields.get(index).id) ? (
+                        <Button
+                          ghost
+                          onClick={() => {
+                            fields.remove(index);
+                          }}
+                        >
+                          <img name="remove" src={TRASHCAN} alt="Remove MineType" />
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:152]: Irrelevant alt text

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 5]
```diff
@@ -48,211 +87,264 @@ const renderContacts = ({
...
+                        component={RenderSelect}
+                        data={filteredRelationships}
+                        validate={[required]}
+                      />
+                      {confirmedContacts?.includes(fields.get(index).id) && (
+                        <Field
+                          usedOptions={rolesUsedOnce}
+                          id={`${field}.party_guid`}
+                          name={`${field}.party_guid`}
+                          label="Selected CORE contact"
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:219]: "Selected CORE contact" -->"Selected Core Contact" or "Core Contact"?

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 6]
```diff
@@ -48,211 +87,264 @@ const renderContacts = ({
...
+                          component={RenderSelect}
+                          data={selectedData}
+                          // disabled
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:222]: Remove if not needed

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 7]
```diff
@@ -48,211 +87,264 @@ const renderContacts = ({
...
+        initialValues: {
+          ...this.state.selectedNOWContact.party?.address?.[0],
+          ...this.state.selectedNOWContact.party,
+        },
       },
       content: modalConfig.ADD_QUICK_PARTY,
     });
   };
 
+  handleReSet = () => {
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:329]: handleReSet --> handleReset

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 8]
```diff
@@ -262,10 +354,82 @@ export class VerifyNoWContacts extends Component {
...
+    });
+    e.preventDefault();
+    this.props.clearAllSearchResults();
+    this.props.setSelectedRows([]);
+    return this.props.fetchSearchResults(searchTerm, "party").then(() => {
+      this.setState({ isLoading: false });
+    });
+  };
+
+  handlSimpleSearch = (searchTerm) => {
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:383]: handlSimpleSearch --> handleSimpleSearch

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 9]
```diff
@@ -279,23 +443,179 @@ export class VerifyNoWContacts extends Component {
...
+  }
+
+  renderCoreContacts = () => {
+    return (
+      <Col span={8}>
+        <Row className="contact-rows">
+          <Col span={24} style={{ minHeight: "130px" }}>
+            <h3>Contact Detail</h3>
+            <p>
+              Use this information to determine if this is the correct contact to use in CORE for
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:473]: We should change all instances of "CORE" to "Core"? I don't think we stylize the name in all caps like that, do we? We should be consistent

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js [hunk 10]
```diff
@@ -279,23 +443,179 @@ export class VerifyNoWContacts extends Component {
...
+
+  renderSearchResults = () => {
+    return (
+      <Col span={8}>
+        <Row className="contact-rows">
+          <Col span={24} style={{ minHeight: "130px" }}>
+            <h3>Matching Contact Options</h3>
+            <p>
+              Click on a contact(s) below to see their detailed information in the &quot;Contact
+              detail&quot; column. If you cannot find a match, you can create a new contact.
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:549]: "Contact detail" --> "Contact Detail"

**Review by exzizt — COMMENTED** [review]
image
Looks really good so far! I've looked at all the code. I'll do more thorough testing on Monday.

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js (hunk 1 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:1]: 😅

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js (hunk 9 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js:473]: You are correct. I will remove all `CORE` instances.

File: services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js (hunk 3 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/VerifyNoWContacts.js]: updated!

**Review by exzizt — APPROVED** [review]
✔️

**Pull request merged**